File: DB/10월 2주차/피보나치수.js
function solution(n) {
  let fibonacci = [0, 1];

  //n번쨰 피보나치까지만 구하자
  for (let i = 0; i < n - 1; i++) {
    fibonacci.push((fibonacci[i] + fibonacci[i + 1]) % 1234567);
  }
  return fibonacci[n];
}

File: MK/10월 2주차/피보나치수.java
/**
 * 피보나치 수
 * https://programmers.co.kr/learn/courses/30/lessons/12945
 */
class Solution {
    public int solution(int n) {
        int[] dp = new int[n+1];
        dp[1] = 1;

        for (int i=2; i<=n; i++) {
            dp[i] = (dp[i-1] + dp[i-2])%1234567;
        }

        return dp[n];
    }
}